SeQuant/core/expr.hpp

```diff
@@ -369,25 +392,25 @@ class Expr : public std::enable_shared_from_this<Expr>,
   /// @return true if this object is of type @c T
   template <typename T>
   bool is() const {
-    if constexpr (std::is_same_v<std::decay_t<T>, Expr>)
+    if constexpr (is_expr_v<T>)
       return true;
     else
-      return this->type_id() == get_type_id<std::decay_t<T>>();
+      return this->type_id() == get_type_id<remove_cvref_t<T>>();
   }
 
   /// @tparam T an Expr type
   /// @return this object cast to type @c T
   template <typename T>
   const T &as() const {
-    assert(this->is<std::decay_t<T>>());  // so that as<const T>() works fine
+    assert(this->is<T>());
```

Comment:
Note that the `is<T>()` will perform the cv_ref removal, so it is not needed here